### PR TITLE
Detect ign instead of using cmake module to check for ignition-tools

### PR DESCRIPTION
**Pull request by j-rivero**
Part of ignition-tooling/release-tools#472

I don't detect the conditional installation of the ign submodule, the PR just
changed the way of detecting ign for testing.

### CODE REVIEW

File: test/test_config.h.in [hunk 1]
```diff
@@ -22,7 +22,6 @@
 #define PROJECT_BINARY_PATH "${PROJECT_BINARY_DIR}"
 #define IGN_TEST_LIBRARY_PATH "${PROJECT_BINARY_DIR}/src:"\
   "@IGNITION-MSGS_LIBRARY_DIRS@:"
-#define IGN_PATH "@IGNITION-TOOLS_BINARY_DIRS@"
```
Comment (mjcarroll) [inline test/test_config.h.in:25]: I don't know if you removed this intentionally, but `src/ign_TEST.cc` depends on this variable being set.

Comment (caguero) [reply test/test_config.h.in:25]: I think `find_program` returns the path in the variable, so I think we could use `HAVE_IGN_TOOLS` to get the path. Then, either we keep this variable using `HAVE_IGN_TOOLS`, or we remove it and use `HAVE_IGN_TOOLS` instead.

Comment (caguero) [reply test/test_config.h.in:25]: Friendly ping @j-rivero

Comment (j-rivero) [reply test/test_config.h.in:25]: > I think `find_program` returns the path in the variable, so I think we could use `HAVE_IGN_TOOLS` to get the path. Then, either we keep this variable using `HAVE_IGN_TOOLS`, or we remove it and use `HAVE_IGN_TOOLS` instead.

You are right, at least my test on Linux is working just fine. 69bdd88

**Comment by mjcarroll** [comment]
@osrf-jenkins retest this please

**Review by caguero — APPROVED** [review]

**Review by mjcarroll — APPROVED** [review]

**Review by chapulina — CHANGES_REQUESTED** [review]
CI is still unhappy

```
   [ RUN      ] ignTest.ServiceRequest
  /github/workspace/tools/ign_TEST.cc:367: Failure
  Value of: "data: " + value + "\n\n"
    Actual: "data: 10\n\n"
  Expected: output
  Which is: "sh: 1: /usr/bin/ign/ign: not found\n"
```

**Comment by chapulina** [comment]
Closing and opening to retrigger CI

**Review by chapulina — APPROVED** [review]
Fixed tests in [382e023](https://github.com/gazebosim/gz-transport/pull/249/commits/382e023b927c2fcc3cbb06182fa2b1b98c346eb5)

**Pull request merged**